mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/JobArtifactRouteHandler.java

```diff
@@ -36,4 +37,9 @@ public interface JobArtifactRouteHandler {
      * Search job artifact names by given prefix. Returns only the names for faster lookups.
      */
     CompletionStage<JobArtifactProto.ListJobArtifactsByNameResponse> listArtifactsByName(final JobArtifactProto.ListJobArtifactsByNameRequest request);
+
+    /**
+     * Upsert mapping resource cluster - job artifact to be cached.
+     */
+    CompletionStage<BaseResponse> upsertJobArtifactToCache(JobArtifactProto.JobArtifactToCacheRequest request);
```

Comment:
Do you think the “which artifacts to cache?” semantics should be under the ResourceCluster component instead of the ResourceCluster route handler endpoint?